server/src/test/scala/org/http4s/server/middleware/AutoSlashSuite.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s.server.middleware
+
+import cats.effect._
+import org.http4s.Uri.uri
+import org.http4s.syntax.all._
+import org.http4s.server.{MockRoute, Router}
+import org.http4s.{Http4sSuite, HttpRoutes, Request, Status}
+
+class AutoSlashSuite extends Http4sSuite {
+  val route = MockRoute.route()
+
+  val pingRoutes = {
+    import org.http4s.dsl.io._
+    HttpRoutes.of[IO] { case GET -> Root / "ping" =>
+      Ok()
+    }
+  }
+
+  test("Auto remove a trailing slash") {
+    val req = Request[IO](uri = uri("/ping/"))
+    route.orNotFound(req).map(_.status).assertEquals(Status.NotFound) *>
+      AutoSlash(route).orNotFound(req).map(_.status).assertEquals(Status.Ok)
+  }
+
+  test("Match a route defined with a slash") {
+    AutoSlash(route)
+      .orNotFound(Request[IO](uri = uri("/withslash")))
+      .map(_.status)
+      .assertEquals(Status.Ok) *>
```

Comment:
What happens if you forget the `*>`?  A warning on discarded values?

Reply:
I didn't see the warning and it is easy to miss. I thought `sbt-tpolecat` would include a flag to warn about this.

Reply:
That's scary.  I'm worried about losing assertions.  It has been one of my problems with these tests-as-values frameworks, even though I think it's a net win.